### PR TITLE
Created KuberenetesResource implementations for EventType

### PR DESCRIPTION
**Pull request by Cali0707**
Part of #3181 

## Proposed Changes

- Create EventType KubernetesResource implementations so that we can have proper typing when implementating the AutoCreate features

### CODE REVIEW

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/KReference.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+
+import java.util.Objects;
+
+@JsonDeserialize
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"apiVersion", "kind", "name", "namespace"})
+public class KReference implements KubernetesResource {
```
Comment (matzew) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/KReference.java:14]: good ol' JavaBeans :smile:

**Comment by Cali0707** [comment]
@pierDipi I am getting the following warnings which cause the unit tests to fail, but I believe I added the correct license header. Any ideas what is going on?

> Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeStatus.java
Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventType.java
Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeSpec.java
Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/KReference.java
Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeBuilder.java
Warning:  Missing header in: /home/runner/work/eventing-kafka-broker/eventing-kafka-broker/data-plane/core/src/test/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeTest.java

File: vendor/knative.dev/eventing/hack/update-cert-manager.sh
Comment (pierDipi) [inline vendor/knative.dev/eventing/hack/update-cert-manager.sh:1]: @Cali0707 I'm not sure about this change /cc @creydr

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeStatus.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import java.util.List;
+import java.util.Objects;
+
+@JsonDeserialize
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"annotations", "conditions", "observedGeneration"})
+public class EventTypeStatus {
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeStatus.java:29]:
```suggestion
public class EventTypeStatus {

```

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeSpec.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+import java.util.Objects;
+
+@JsonDeserialize
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"broker", "reference", "description", "schema", "schemaData"})
+public class EventTypeSpec implements KubernetesResource {
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeSpec.java:29]:
```suggestion
public class EventTypeSpec implements KubernetesResource {

```

File: vendor/knative.dev/eventing/hack/update-cert-manager.sh
Comment (Cali0707) [reply vendor/knative.dev/eventing/hack/update-cert-manager.sh:1]: @pierDipi my `Verify Deps an Codegen` tests were failing so I ran `./hack/update-deps.sh` and it made this change - should we change anything in that script?

Comment (Cali0707) [reply vendor/knative.dev/eventing/hack/update-cert-manager.sh:1]: This change is now reverted

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**